### ISSUE TITLE
Active index not working as expected in repeater component

### ISSUE THREAD

**Issue by blackbaud-vimal-kumar** [blackbaud/skyux-split-view#13]
On updating repeater list, the active index is being lost in following cases:
1. If active index does not change
2. If activeIndex is changed and list is updated, we have to use a settimeout to set it, as being too close creates problem.
3. I think we are also loosing active index any time we re-render the repeater list.

**Comment by blackbaud-vimal-kumar** [comment]
More info:
https://blackbaud.slack.com/archives/C3TQ4C0KG/p1575891593223200?thread_ts=1575567912.181100&cid=C3TQ4C0KG

**Comment by Blackbaud-AlexKingman** [comment]
Hi @blackbaud-vimal-kumar, thanks for submitting.

We're having trouble replicating this issue. We're able to dynamically update the list and it retains the activeIndex correctly. I'm wondering if there may be some trouble with the asynchronous nature of how your repeater items are getting updated.

I took a look at your voulenteer SPA code, and it looks like the `teams` object (which populates your repeater), is being updated by an Observable.

```
this.activeIndex = this.teams.indexOf(team);
```

Is it possible the observable isn't completed when the above expression fires? If `this.teams` is empty, it could be the reason your `activeIndex` gets lost. Is there a way you can set `activeIndex` after the observable finishes?

If I'm misunderstanding what's happening here, please let me know. I'd love to run your SPA locally to debug further, but may need access to some security groups to get it to run.

**Comment by blackbaud-vimal-kumar** [comment]
this.teams.push(this.allTeamMenuItem); // code responsible for pushing the new list
this.activeIndex = this.teams.indexOf(team); // code responsible for setting the active index

are happening on the same observable subscribe, in above order.
Issue is still persist, if I would remove the serTimeout, it simply does not highlight anything

**Comment by Blackbaud-AlexKingman** [comment]
@blackbaud-vimal-kumar Just curious... if you remove the `setTimeout` and debug this line:
```
this.activeIndex = this.teams.indexOf(team)
```
What does `this.activeIndex` get set to?

**Comment by blackbaud-vimal-kumar** [comment]
If I remove this setTimeout, it results in nothing being highlighted.

**Comment by Blackbaud-AlexKingman** [comment]
That's my suspicion as well. Can you debug that line for me and let me know what `this.activeIndex` gets set to? I'm guessing it might be `-1` ?

**Comment by blackbaud-vimal-kumar** [comment]
Actually it is returning the index correctly, but the list is not lighting.

Again there are two separate issues here.
The index being lost without setimeout and index being lost when it does not change but the list reloads

Created a video: 
https://blackbaud.slack.com/archives/C3TQ4C0KG/p1582805817083100?thread_ts=1546544111.050500&cid=C3TQ4C0KG

**Comment by Blackbaud-AlexKingman** [comment]
@blackbaud-vimal-kumar Could you please try upgrading your `@skyux/lists` package to the latest? We believe this is a duplicate of blackbaud/skyux-lists/#63 and was fixed in version `3.7.3`.

**Comment by Blackbaud-AlexKingman** [comment]
Confirmed fix by OP. Closing.
